presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -294,6 +294,7 @@ public final class SystemSessionProperties
     public static final String REWRITE_CONSTANT_ARRAY_CONTAINS_TO_IN_EXPRESSION = "rewrite_constant_array_contains_to_in_expression";
     public static final String INFER_INEQUALITY_PREDICATES = "infer_inequality_predicates";
     public static final String ENABLE_HISTORY_BASED_SCALED_WRITER = "enable_history_based_scaled_writer";
+    public static final String USE_PARTIAL_AGGREGATION_HISTORY = "use_partial_aggregation_history";
```

Comment:
that makes sense.
The only caveat here is that this optimization was already history-based, but now we use the statistics from the partial aggregation instead of the final aggregation. I added the flag to avoid possible regressions and be able to gradually deploy.
If you have more name suggestions let me know :)

Reply:
Make sense. Let's keep the name. However It may still be worth to have a consistent session property name (`use_partial_aggregation_history`) and config property name (`optimizer.use-partial-aggregation-history`)